### DISCUSSION TITLE
Could not see any files in the explorer...

### DISCUSSION THREAD

**Discussion by miketaochina** [coder/code-server#3607] [Q&A]
Dear: 

I have set up everything for code-server, and opened it in safari, but it could not show any files in the explorer. 

I checked the `code-server.log`, it showed:
```
[node.js fs] readdir with filetypes failed with error:  [Error: EPERM: operation not permitted, scandir '/Users/name/Desktop/mitStudy/Courses/Network Security/python_courses/Personal'] {
  errno: -1,
  code: 'EPERM',
  syscall: 'scandir',
  path: '/Users/name/Desktop/mit/Study/Courses/Network Security/python_courses/Personal'
}
```
anyone can help please? thanks.

Best regards 
Mike

**Comment by jsjoeio** [comment]
Hmm...it looks like it wasn't able to read your directories due to permissions? Where is code-server running and how did you install it?

**Reply by miketaochina** [reply]
Dear: 

```
it looks like it wasn't able to read your directories due to permissions?
```
Yes, I think so, but I do not know why...

```
how did you install it?
```
I used `brew install code-server`

```
Where is code-server running
```
I do not know, how could I check that?
But I found a code-server folder in my Mac: `/usr/local/Cellar`.

Thanks a lot,
Best regards,

Mike

**Reply by jsjoeio** [reply]
That helps! Can you try [this](https://github.com/cdr/code-server/issues/2420#issuecomment-741494971)?

**Reply by miketaochina** [reply]
Dear: 

I have checked this, and my terminal have already had access to those folders.
<img width="780" alt="access to those folders" src="https://user-images.githubusercontent.com/69522499/122266082-b4b64000-ced9-11eb-9c60-afe01c23d948.png">

Is there anything else I can do? I really want it works.

Many thanks,


Best regards,

Mike

**Reply by jsjoeio** [reply]
Hmm...I'm not sure why it doesn't have the right permissions. What browser are you running it in? Can you try doing [this](https://stackoverflow.com/a/65710222/3015595) but for the browser you're using?

**Reply by miketaochina** [reply]
Dear: 

```
What browser are you running it in?
```
I use safari from both my Mac(Catalina)(as my code-server' main server) and iPad Pro.  

I have tried [this](https://stackoverflow.com/a/65710222/3015595), and I have given the full access to `Files and Folders` for both `safari` and `terminal`, you can see for the pictures, they are in Chinese, but you know what I mean.
<img width="780" alt="截屏2021-06-16 20 05 25" src="https://user-images.githubusercontent.com/69522499/122271560-d286a380-cedf-11eb-9761-fdb7d6e6d879.png">

<img width="780" alt="截屏2021-06-16 20 05 29" src="https://user-images.githubusercontent.com/69522499/122271656-f0ec9f00-cedf-11eb-81ba-43e52efe6ad9.png">

And I also restart `code-server` and `safari`, still does not work.

Is that possible for you to remote to my Mac and try something?

Many thanks,
Best regards,

Mike

**Reply by miketaochina** [reply]
Also: 

Every time I refresh the code-server page, it will show me this:
```
code-server is being accessed over an insecure domain. Web views, the clipboard, and other functionality will not work as expected.
```
Here is how it looks like:  
<img width="557" alt="截屏2021-06-16 20 02 49" src="https://user-images.githubusercontent.com/69522499/122273287-8f2d3480-cee1-11eb-999d-953181161f0e.png">

Mike

**Reply by jsjoeio** [reply]
What version of macOS are you running?

Can you check the permissions of that directory with `ls -ld /Users/name/Desktop/mit/Study/Courses/Network Security/python_courses/Personal` and show what it outputs?

Last thing might be to uninstall and reinstall code-server:
```
brew uninstall code-server

brew install code-server
```

**Reply by miketaochina** [reply]
Dear: 

```
What version of macOS are you running?
```
I am using Mac OS `Catalina(10.15.7)`. MacBook Pro (15-inch, 2018)

I have tried to check the permissions of that directory, and it showed me something interesting:  

When I try:  `ls -ld /Users/mike/Desktop/MT/Study`, it showed me: `drwxr-xr-x@ 14 mike  staff  448  6 16 19:51 /Users/mike/Desktop/MT/Study`


When I try: `ls -ld /Users/mike/Desktop/MT/Study/Courses`, it showed me: `drwxr-xr-x@ 8 mike  staff  256  6 16 19:51 /Users/mike/Desktop/MT/Study/Courses`


When I try: `ls -ld /Users/mike/Desktop/MT/Study/Courses/Network Security`, it showed me: 
```
ls: /Users/mike/Desktop/MT/Study/Courses/Network: No such file or directory
ls: Security: No such file or directory
```

<img width="1102" alt="截屏2021-06-18 18 13 01" src="https://user-images.githubusercontent.com/69522499/122589688-e0affd80-d060-11eb-8494-c671edd8a8ac.png">

You see that `Network` and `Security` is separated...If I keep trying more path, It gives me the same results, I do not get it. 
But I can see and edit all the files and folders in VS code locally... 

Many thanks,
Best regards,

Mike

**Reply by jsjoeio** [reply]
> When I try: ls -ld /Users/mike/Desktop/MT/Study/Courses/Network Security, it showed me

This is because "Network Security" has a space in it. You need to either escape it or put the name in quotes. See this example:
![image](https://user-images.githubusercontent.com/3806031/122595744-be48c080-d01d-11eb-828e-5329ffb0835c.png)

Can you try running:
```shell
ls -ld "/Users/name/Desktop/mit/Study/Courses/Network Security/python_courses/Personal"
```

(My fault, I should have put it in quotes when I first asked)

**Reply by jsjoeio** [reply]
cc @oxy or @code-asher in case you have other ideas for solving this

**Reply by code-asher** [reply]
Hm it does seem like code-server doesn't have permissions to that directory. Running that command seems like a good way to test that theory.

**Reply by miketaochina** [reply]
Dear: 

It worked when try: `ls -ld "/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/Personal"`   

<img width="1272" alt="截屏2021-06-19 02 44 41" src="https://user-images.githubusercontent.com/69522499/122626901-03680380-d0ad-11eb-99de-6813cfa5a5d5.png">

You are awesome, man! so what's next then? I still need to see and edit files and folders in the explorer...

And how I should escape the `space` from `Network Security`? Make it into `Network_Security`. Should I change it in macOS files system or change it in VScode?

Many thanks,
Best regards,

Mike

**Reply by code-asher** [reply]
You don't need to escape the space, the quote marks in the `ls` command handle it for you!

This is strange though. It looks like you have permissions to the directory. Are you able to create files from the terminal there? For example:

```bash
cd "/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/Personal"
touch test
```

If that works, try:

```bash
code-server "/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/Personal"
```

**Reply by miketaochina** [reply]
Dear:  
```
Are you able to create files from the terminal there?
```
I have tried: `cd "/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/Personal"
touch test` and it showed me nothing, but when I go to Mac finder, I can see that the files are `created`.
<img width="327" alt="截屏2021-06-22 18 58 44" src="https://user-images.githubusercontent.com/69522499/122968524-ec1a6600-d38b-11eb-8b5a-c42937fc69d2.png">

Then I have tried: `code-server "/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/Personal"`, it showed me:  
```
(base) YEMBP:Personal mike$ code-server "/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/Personal"
[2021-06-21T19:20:43.657Z] info  code-server 3.10.2 387b12ef4ca404ffd39d84834e1f0776e9e3c005
[2021-06-21T19:20:43.658Z] info  Using user-data-dir ~/.local/share/code-server
[2021-06-21T19:20:43.669Z] info  Using config file ~/.config/code-server/config.yaml
[2021-06-21T19:20:43.669Z] info  HTTP server listening on http://192.168.1.175:8080 
[2021-06-21T19:20:43.669Z] info    - Authentication is enabled
[2021-06-21T19:20:43.669Z] info      - Using password from ~/.config/code-server/config.yaml
[2021-06-21T19:20:43.669Z] info    - Not serving HTTPS 
[IPC Library: Pty Host]  INFO Persistent process "2": Replaying 219 chars and 1 size events
```
<img width="1092" alt="截屏2021-06-22 18 50 08" src="https://user-images.githubusercontent.com/69522499/122967912-4535ca00-d38b-11eb-8b40-99f16af15532.png">
And suddenly the code-server works, I can see and edit the files and folders from `code-server web`.
And one more question here, how I can run the code I write from `code-server web`? For example: the hello.py in the path: `"/Users/mike/Desktop/MT/Study/Courses/Network Security/python_courses/1DV501/1DV501/yt222ar_assign1/hello.py"`

Many thanks,
Best regards,

Mike

**Reply by jsjoeio** [reply]
> And one more question here, how I can run the code I write from `code-server web`?

You should be able to open that folder in code-server, open the Integrated Terminal and then run `python hello.py`

**Reply by miketaochina** [reply]
Dear:  
```
open that folder in code-server
```
I have done this and tried both `python hello.py` and `python3 hello.py`, but none of them worked, they showed: `can't open file 'hello.py': [Errno 2] No such file or directory`.   

<img width="1680" alt="截屏2021-06-22 22 33 01" src="https://user-images.githubusercontent.com/69522499/122995747-47f3e780-d3aa-11eb-9c55-7493cabc3b10.png">

And unfortunately when I writing the message, code-server web stoped working again(could not see anything again), so I tried above thing in vscode locally.
So I am think to escape the `space` from `Network Security`, make it into` Network_Security`. Should I change it in `macOS files system` or change it in `VScode`?
Many thanks,
Best regards,

Mike

**Reply by jsjoeio** [reply]
>  have done this and tried both

From the screenshot it looks like you're Terminal is in a different folder than `hello.py` is in? Which is why that wouldn't work. You need to `cd` into that folder.  Looks like from the file explorer you're in `yt222ar_assign1` but the terminal is in `Courses`? Hard to tell.

Inside terminal, run `pwd` to see the working directory. If it shows `yt222ar_assign1` then you are in the right place and can run `python hello.py`. If it isn't, then you need to `cd` into that directory and then you can run `python hello.py`.

> Should I change it in macOS files system or change it in VScode

Either works! I would use whatever you're more comfortable with!
